### ISSUE TITLE
Major changes to the code

### ISSUE THREAD

**Issue by BelieveF** [BingfengYan/CO-MOT#19]
Thank you for your excellent work! I wonder if the biggest change (i.e. COLA) in your project code from MOTR is in the def match_for_single_frame(self, outputs: dict) method? Have you made any changes to deformable_transformer_plus.py? I would appreciate it if you could answer!

**Comment by fengxiuyaun** [comment]
Yes, we changed the match_for_single_frame function for COLA. It is very simple, but effective.

**Comment by BelieveF** [comment]
> Yes, we changed the match_for_single_frame function for COLA. It is very simple, but effective.

Thank you for your answer, your work is very helpful to me!

**Comment by yml2002** [comment]
Thank you for your excellent work! I would like to know if I can use your code to help me infer a new video, specifically to detect the characters in this video and track them. Is your work solely focused on the tracking module? Is the detection module you use YOLOv8? I noticed in your proposal that each video requires a corresponding TXT file. Does this mean that I can not use your method to detect a new video that is not annotated?

**Comment by fengxiuyaun** [comment]
> Thank you for your excellent work! I would like to know if I can use your code to help me infer a new video, specifically to detect the characters in this video and track them. Is your work solely focused on the tracking module? Is the detection module you use YOLOv8? I noticed in your proposal that each video requires a corresponding TXT file. Does this mean that I can not use your method to detect a new video that is not annotated?

This is a End-to-End tracking model, without detection module. You can run it on your videos.

**Comment by yml2002** [comment]
Thank you for your response. Can I use your tracking module, combined with other detection modules, to implement the functionality of detecting a new video, thereby generating content in the same format as "<frame>, <id>, <bb_left>, <bb_top>, <bb_width>, <bb_height>, <conf>, -1, -1, -1" found in dancetrack000x.txt from the DanceTrack Test? Or is this functionality generated by the detection module's txt file? I noticed that your code utilizes the load_img_from_file function, which requires a corresponding txt file for each image. Is this used for comparing the results obtained by the model with the actual results? If so, I would like to save the results obtained from your model directly.

**Comment by fengxiuyaun** [comment]
> Thank you for your response. Can I use your tracking module, combined with other detection modules, to implement the functionality of detecting a new video, thereby generating content in the same format as ", , <bb_left>, <bb_top>, <bb_width>, <bb_height>, , -1, -1, -1" found in dancetrack000x.txt from the DanceTrack Test? Or is this functionality generated by the detection module's txt file? I noticed that your code utilizes the load_img_from_file function, which requires a corresponding txt file for each image. Is this used for comparing the results obtained by the model with the actual results? If so, I would like to save the results obtained from your model directly.

No, this model does not require detection modules. You can refer to the paper for more details.
